core-api/src/main/java/com/optimizely/ab/odp/ODPManager.java

```diff
@@ -0,0 +1,61 @@
+/**
+ *
+ *    Copyright 2022, Optimizely
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.odp;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class ODPManager {
+    private volatile ODPConfig odpConfig;
+    private final ODPSegmentManager segmentManager;
+    private final ODPEventManager eventManager;
+
+    public ODPManager(@Nonnull ODPConfig odpConfig, @Nonnull ODPApiManager apiManager) {
+        this(odpConfig, new ODPSegmentManager(odpConfig, apiManager), new ODPEventManager(odpConfig, apiManager));
+    }
+
+    public ODPManager(@Nonnull ODPConfig odpConfig, @Nonnull ODPSegmentManager segmentManager, @Nonnull ODPEventManager eventManager) {
```

Comment:
Customization of segmentCache size/timeout and segmentCache itself as well?

Reply:
We also need to support for "disable" entire ODP functions.

Reply:
There are two constructors.
1. First one only takes the two mandatory arguments of `ODPConfig` and `ApiManager` and initializes everything else on default values. 
2. If you want to customize anything at all, use the second constructor which expects you to pass everything from the outside. This is where you will have a chance to initialize your own `SegmentManager` with your own `cacheOptions`.

I will recommend to keep these two options for now. We can provide other combinations at the level of factory methods to create the `SDK` instance later as needed. What do you think?

Reply:
I am thinking of handling disable / enable at `optimizely` level. When enabled, ODP object will be there and it will always work. When disabled, there will be no ODP instance at all. What do you suggest?

Reply:
> There are two constructors.
> 
> 1. First one only takes the two mandatory arguments of `ODPConfig` and `ApiManager` and initializes everything else on default values.
> 2. If you want to customize anything at all, use the second constructor which expects you to pass everything from the outside. This is where you will have a chance to initialize your own `SegmentManager` with your own `cacheOptions`.
> 
> I will recommend to keep these two options for now. We can provide other combinations at the level of factory methods to create the `SDK` instance later as needed. What do you think?

I see adjusting cache size and timeout more likely options for clients rather than full Cache and OdpManager customazations. Providing that path with factory will be also a good solution too.

Reply:
> I am thinking of handling disable / enable at `optimizely` level. When enabled, ODP object will be there and it will always work. When disabled, there will be no ODP instance at all. What do you suggest?

I was concerned about "if odpManager == null" checking in many places we need odpManager. We can try wrap those checking into OdpManager. Other than that, disabling at the top level is also a good option too.

Reply:
I would advise to defer this discussion till the top level PR. We can then see if it looks good at top level other wise i will add this to ODPManager. what do you say?